Baseline upgrades (#1231)
umbraco/UmbracoDocs: Umbraco-Cloud/Getting-Started/Baselines/Upgrading-Child-Projects/index.md

4. Follow the process as the upgrades are pushed to the child projects one by one
5. When the Child Projects *turn green* the upgrade was a success

![Upgrade Child Projects](images/manage-baseline-children.gif)

## Deploying Minor upgrades to Child projects

To do a minor upgrade of a Baseline project and its Child projects, the initial task is to run the minor upgrade on the baseline it self. This is done as you'd expect in the Umbraco Cloud portal by clicking the "Upgrade available!" button on the Development environment. 

Following [Minor upgrades](../../Upgrades/#minor-version-upgrades) guide and process will leave the Baseline project upgraded to the latest version, while all the children are untouched for now.

Once the upgrade has been verified on the Baseline project, and you've made sure everything works as expected, you should go to the **Manage Child Projects** page on the Baseline. 

> On this page you'll notice that all the children will be marked as if they have a pending upgrade. At this point you are able to select one or more projects to upgrade. 

Reviewer: ```suggestion
On this page you'll notice that all the children will be marked with pending upgrades. At this point you are able to select one or more projects to upgrade. 
```

![Minor upgrade labels](images/minor-upgrades.png)

> The upgrade itself will happen, once you click the upgrade button. This will start by triggering the update, where all the files are updated on the children from the baseline. Once the files are in place we also run the upgrade process, making sure that the children are fully upgraded, once done. 

Reviewer: ```suggestion
The upgrade itself will happen, once you click the upgrade button. This will start by triggering the update, where all the files are updated on the children from the baseline. Once the files are in place we also run the upgrade process, making sure that the children are fully upgraded. 
```

> When using the feature, it is important that the Baseline Child projects have been setup following our [best practices for handling config files](../Configuration-files). This means that any changes to the Child project, shoud be applied via a config transform file. The reason is that the Child projects config files will be merged by choosing the parents config files first, to ensure that changes to config files, that has been made in the minor upgrade, will also be applied to the childsites.

Reviewer: "to the child sites."

### Errors while upgrading children from baseline

> If for some reason the update of a child fail, or the child is left in a weird state, where it has some wrong configuration files, it is most likely because the child was unable to be merged properly. When updating Child projects from a Baseline project, a configuration file will be preferred from the child over the one from the baseline. This means that when the update from the baseline to the child runs, the configuration file sometimes won’t be changed. 

Reviewer: "bad state" in stead of "weird state"? :)

> If for some reason the update of a child fail, or the child is left in a weird state, where it has some wrong configuration files, it is most likely because the child was unable to be merged properly. When updating Child projects from a Baseline project, a configuration file will be preferred from the child over the one from the baseline. This means that when the update from the baseline to the child runs, the configuration file sometimes won’t be changed. 

Reviewer: instead of "preferred from":

"the configuration file from the child project will take precedence over the baseline project configuration"

4. Follow the process as the upgrades are pushed to the child projects one by one
5. When the Child Projects *turn green* the upgrade was a success

![Upgrade Child Projects](images/manage-baseline-children.gif)

## Deploying Minor upgrades to Child projects

To do a minor upgrade of a Baseline project and its Child projects, the initial task is to run the minor upgrade on the baseline it self. This is done as you'd expect in the Umbraco Cloud portal by clicking the "Upgrade available!" button on the Development environment. 

Following [Minor upgrades](../../Upgrades/#minor-version-upgrades) guide and process will leave the Baseline project upgraded to the latest version, while all the children are untouched for now.

Once the upgrade has been verified on the Baseline project, and you've made sure everything works as expected, you should go to the **Manage Child Projects** page on the Baseline. 

> On this page you'll notice that all the children will be marked with pending upgrades. At this point you are able to select one or more projects to upgrade. 

Reviewer 2: ```suggestion
On this page you'll notice that all the children now have an available upgrade. At this point you are able to select the projects to upgrade.

At first any pending changes made on the Baseline will be deployed to the child site. Once the changes have been deployed, the child site will be upgraded to the same version as the Baseline site. All the products (CMS, Courier to Deploy and Forms) will be upgraded.
```

I believe it's important to note down that any pending changes made on the Baseline but not yet on the Child site will be at first deployed/synced and the upgrade will start afterwards.
